wlroots-sys/build.rs

```diff
@@ -103,13 +129,399 @@ fn main() {
+    if check_version(
+        "wayland-protocols".to_string(),
+        "--version".to_string(),
+        "0".to_string(),
+        true
+    ) && "wayland-protocols" == command
+    {
+        println!("wayland-protocols found");
+    } else {
+        println!("WRONG version of wayland-protocols or not installed.");
+
+        println!("\nInstallation instructions, install with packet manager or:");
+        println!("git clone https://github.com/wayland-project/wayland-protocols.git");
+    }
+
+    return "".to_string();
+}
+
+fn package_error_common_unstable() -> String {
+
+    if cfg!(feature = "unstable") {
+        println!("Are the following unstable dependencies installed?");
+        println!("libwayland-dev");
+        println!("libudev-dev");
+        println!("libgles2-mesa-dev");
+        println!("libpixman-1-dev");
+        println!("llvm");
+        println!("libxkbcommon-dev");
+        println!("libxkbcommon-dev");
+        println!("libinput-dev");
+        println!("libinput-bin");
+    }
+
+    return "".to_string();
+}
+
+
+// STATIC BUILD CHECK
+#[cfg(feature = "static")]
+fn package_error_static() {
+    if check_version(
+        "pkg-config".to_string(),
+        "--version".to_string(),
+        "0".to_string(),
+        false
+    )
+    {
+        println!("pkg-config found");
+    } else {
+        println!("WRONG version of pkg-config or not installed on system.");
+
+        println!("\nInstallation instructions, install with packet manager");
+        println!("Installation cannot rely on Cargo pkg-config ");
+        exit(2);
+    }
+
+    if check_version(
+        "ninja".to_string(),
+        "--version".to_string(),
+        "1.9.0".to_string(),
+        true
+    )
+    {
+        println!("ninja found");
+    } else {
+        println!("WRONG version of ninja or not installed on system.");
+
+        println!("\nInstallation instructions, install with packet manager or python3-pip if available");
+        exit(2);
+    }
+
+    if check_version(
+        "meson".to_string(),
+        "--version".to_string(),
+        "0.54.0".to_string(),
+        true
+    )
+    {
+        println!("meson found");
+    } else {
+        println!("WRONG version of meson or not installed on system.");
+
+        println!("\nInstallation instructions, install with packet manager or python3-pip if available");
+        println!("STATIC installation cannot rely on Cargo meson ");
+        exit(2);
+    }
+
+    if check_version(
+        "cmake".to_string(),
+        "--version".to_string(),
+        "3.0".to_string(),
+        true
+    )
+    {
+        println!("cmake found");
+    } else {
+        println!("WRONG version of cmake or not installed on system.");
+
+        println!("\nInstallation instructions, install with packet manager or if available");
+        println!("STATIC installation cannot rely on Cargo cmake ");
+        exit(2);
+    }
+
+    if check_version(
```

Comment:
Would prefer to use check_version("clang", "--version", "0.0") but that is pretty minor. Then the function would take &str instead of String.